dove/tests/helper.rs

```diff
@@ -0,0 +1,176 @@
+    execute_dove_at(&base_folder, &["dove", "new", project_name]).unwrap_or_else(|err| {
+        panic!("{}", err);
+    });
+    set_dependencies_local_move_stdlib(&project_folder);
+}
+/// $ dove new ### -d ### -a ### -r ###
+pub fn project_new_with_args(
+    base_folder: &Path,
+    project_folder: &Path,
+    project_name: &str,
+    project_dialect: &str,
+    project_address: &str,
+    project_blockchain_api: &str,
+) {
+    execute_dove_at(
+        &base_folder,
+        &[
+            "dove",
+            "new",
+            project_name,
+            "-d",
+            project_dialect,
+            "-r",
+            project_blockchain_api,
+            "-a",
+            project_address,
+        ],
+    )
+    .unwrap_or_else(|err| {
+        panic!("{}", err);
+    });
+    set_dependencies_local_move_stdlib(&project_folder);
+}
+// @dove build
+pub fn project_build(project_folder: &Path) -> Result<String, String> {
+    execute_dove_at(&project_folder, &["dove", "build"])
+}
+pub fn project_remove(project_folder: &Path) {
+    if project_folder.exists() {
+        remove_dir_all(project_folder).unwrap_or_else(|err| {
+            panic!(
+                "[ERROR] Couldn't delete project directory: {}\r\n[MESSAGE] {}\r\n",
+                project_folder.display(),
+                err.to_string()
+            )
+        });
+    }
+}
+pub fn set_dependencies_local_move_stdlib(project_path: &Path) {
+    let move_stdlib = Path::new(".")
+        .canonicalize()
+        .unwrap()
+        .join("resources/test_move_project");
+    let mut dove_toml_path = project_path.to_path_buf();
+    dove_toml_path.push("Dove.toml");
+    let mut toml_value = read_to_string(&dove_toml_path)
+        .unwrap()
+        .parse::<Value>()
+        .unwrap();
+    {
+        let v = toml_value
+            .get_mut("package")
+            .unwrap()
+            .get_mut("dependencies")
+            .unwrap()
+            .as_array_mut()
+            .unwrap();
+        v.clear();
+        let mut dd = toml::map::Map::new();
+        dd.insert(
+            "path".to_string(),
+            Value::String(move_stdlib.to_str().unwrap().to_string()),
+        );
+        v.push(Value::Table(dd));
+    }
+    write_all(
+        &dove_toml_path,
+        toml::to_string(&toml_value).unwrap().as_str(),
+    )
+    .unwrap();
+}
+pub fn execute_dove_at(project_folder: &Path, args: &[&str]) -> Result<String, String> {
+    execute(args, project_folder.to_path_buf()).map_or_else(
+        |err| {
+            Err(format!(
+                "[COMMAND] {}\r\n[FOLDER] {}\r\n[ERROR] {}\r\n",
+                &args.join(" "),
+                &project_folder.display(),
+                err.to_string()
+            ))
+        },
+        |_| {
+            Ok(format!(
+                "[COMMAND] {}\r\n[FOLDER] {}\r\n",
+                &args.join(" "),
+                &project_folder.display(),
+            ))
+        },
+    )
+}
+pub fn check_dove_toml(
```

Comment:
I meant creating raw string like
```
let expected_contents = r#"
[package]
name = "assets"
dialect = "pont"
dependencies = [
    { path = "./stdlib" }
]
"
```
store it in the constant/variable and compare it with the parsed value using smth along the (preudocode)
```
normalized_expected = expected_contents.regex_str_replace(r"[\s]", " ")
normalized_actual = actual_contents.regex_str_replace(r"[\s]", " ")
assert!(normalized_expected == normalized_actual)
```
(that's what @RIg410 meant)

It will allow to catch a case where there's another key added that is not expected.

Reply:
this needs to be discussed